lib/worker/src/routes/vaults.rs

```diff
@@ -502,6 +527,24 @@ pub async fn write_event(
         }
     };
 
+    // Patch the GCS object with the signature and hash
```

Comment:
shouldn't this logic be after signature verification [L548](https://github.com/tablelandnetwork/basin-provider/pull/19/files#diff-c90b016c5ace95d94873265259c3585094b5a95614490f5d4f222b291836ae7eR548) ?  otherwise, you could be inserting an invalid signature